jina/clients/mixin.py

```diff
@@ -57,18 +60,34 @@ async def _get_results(*args, **kwargs):
             if return_results:
                 return result
 
-        return run_async(
-            _get_results,
-            inputs=inputs,
-            on_done=on_done,
-            on_error=on_error,
-            on_always=on_always,
-            exec_endpoint=on,
-            target_peapod=target_peapod,
-            parameters=parameters,
-            request_size=request_size,
-            **kwargs,
-        )
+        with ExitStack() as stack:
+            if (
+                not hasattr(self.args, 'infrastructure')
+                or self.args.infrastructure == InfrastructureType.JINA
+            ):
+                pass
+            elif self.args.infrastructure == InfrastructureType.K8S:
+                stack.enter_context(
+                    kubernetes_tools.get_port_forward_contextmanager(
+                        self.args.name, self.port_expose
+                    )
+                )
+            else:
+                raise NotImplemented(
+                    f'Infrastructure type {self.args.infrastructure} unknown.'
+                )
```

Comment:
I would change this to
```suggestion
        context_mgr = nullcontext()
        if self.args.infrastructure == InfrastructureType.K8S:
            context_mgr = kubernetes_tools.get_port_forward_contextmanager(
                self.args.name, self.port_expose
            )

        with context_mgr:
            ...
```

Reply:
thanks, changed. Much better